Update CHANGELOG.md for v1.12.0 (#2746)
bottlerocket-os/bottlerocket: CHANGELOG.md

# v 1.12.0 (2022-01-24)

> ## OS Changes

Reviewer: systemd update is worth a special call-out

> * Add patch to enable strict aliasing in dbus-broker ([#2730])

Reviewer: this could go in build changes instead, since it was related to the SDK update
```suggestion
* Disable strict aliasing for c-utf8 library in dbus-broker ([#2730])
```

* Add `/sys/firmware` to privileged mounts in host-ctr ([#2714])

> * Use user provided registry credentials for public.ecr.aws in host-ctr ([#2676])

Reviewer: ```suggestion
* Use user-provided registry credentials for public.ecr.aws in host-ctr ([#2676])
```

> * Masked paths list built dynamically in host-ctr ([#2637])

Reviewer: ```suggestion
* Build masked paths list dynamically in host-ctr ([#2637])
```

* Enable EFI option in systemd ([#2714])
* Allow simple enums as map keys in datastore ([#2687])
* Add method for fetching hostname in imdsclient ([#2647])

> * Add version 3 of network configuration to netdog ([#2596])
> * Add vlan structs for network configuration to netdog ([#2596])
> * Add bonding structs for network configuration to netdog ([#2596])

Reviewer: nit: maybe
```suggestion
* Add support for bonding and VLANs in `net.toml` ([#2596])
```

> * Migrator keeps only one intermediate datastore ([#2589])

Reviewer: ```suggestion
* Keep only one intermediate datastore during migration ([#2589])
```

* Widen access to filesystem relabel in SELinux policy ([#2738])
* Update hotdog to 1.05 ([#2728])
* Update third party packages and dependencies ([#2717], [#2718])
* Update eksd ([#2690], [#2693], [#2694], thanks @rcrozean)

## Orchestrator Changes

### Kubernetes

* Add support for Kubernetes 1.25 variants ([#2699])

> * Restrict access to tls private key in kubelet ([#2639])

Reviewer: The private key was already restricted; the actual effect was to remove the restrictions on reading public certificates to fix kube-bench: 
```suggestion
* Allow access to public kubelet certificates ([#2639])
```

* During kubelet prestart, skip pause image pull if image exists ([#2587])

> * Start kubelet.service if warm-pool-wait service is running ([#2562])

Reviewer: This commit message is a bit unclear, maybe:
```suggestion
* Delay kubelet.service until after warm-pool-wait service runs ([#2562])
```

Reviewer 2: Could perhaps also focus on the effect, "Add support for AWS Autoscaling Warm Pools"

## Platform Changes

### Metal

> * Include MAC address in lease files ([#2650])

Reviewer: nit: this ended up getting reverted

Author: Added callout for systemd update

> ### VMWare

Reviewer 3: s/VMWare/VMware/

> * Downgrade warning about inability to iopl to debug ([#2732])

Reviewer: ```suggestion
* Downgrade iopl warning when fetching guestinfo in `early-boot-config` ([#2732])
```

## Build Changes

> * Alias warnings are now treated as errors ([#2730])

Reviewer: ```suggestion
* Treat alias warnings as errors ([#2730])
```

* `source_date_epoch_from_changelog` turned off, removing "missing changelog" warning in build ([#2730])

> * Bottlerocket SDK version updated to 0.29.0 ([#2730])

Reviewer: ```suggestion
* Update Bottlerocket SDK version to 0.29.0 ([#2730])
```

> * Improved error messages for publish-ami command ([#2695])

Reviewer: ```suggestion
* Improve error messages for publish-ami command ([#2695])
```

* Disallow private AMIs in public SSM parameters ([#2680])

> * `start-local-vm` image selection reworked to use `latest` symlink ([#2696])

Reviewer: ```suggestion
* Rework `start-local-vm` image selection to use `latest` symlink ([#2696])
```

# v 1.12.0 (2022-01-24)

> ## OS Changes

Reviewer 4: We should add a line item for updating the host containers https://github.com/bottlerocket-os/bottlerocket/pull/2739

* Disable strict aliasing for c-utf-8 library strict aliasing in dbus-broker ([#2730])
* Add `/sys/firmware` to privileged mounts in host-ctr ([#2714])
* Use user-provided registry credentials for public.ecr.aws in host-ctr ([#2676])
* Build masked paths list dynamically in host-ctr ([#2637])
* Enable EFI option in systemd ([#2714])
* Allow simple enums as map keys in datastore ([#2687])
* Add method for fetching hostname in imdsclient ([#2647])
* Add support for bonding and VLANS in `net.toml` ([#2596])
* Keep only one intermediate datastore during migration ([#2589])
* Widen access to filesystem relabel in SELinux policy ([#2738])
* Update hotdog to 1.05 ([#2728])
* Update systemd to 250.9 ([#2718])
* Update third party packages and dependencies ([#2588], [#2717])
* Update eksd ([#2690], [#2693], [#2694], thanks @rcrozean)

> ## Orchestrator Changes

Reviewer 5: The OCI Defaults feature is important, please include. I guess it goes in the Kubernetes section.

https://github.com/bottlerocket-os/bottlerocket/pull/2697

### Kubernetes

* Add support for Kubernetes 1.25 variants ([#2699])
* Allow access to public kubelet certificates ([#2639])
* During kubelet prestart, skip pause image pull if image exists ([#2587])
* Delay kubelet.service until after warm-pool-wait service runs ([#2562])

## Platform Changes

### VMware

* Downgrade iopl warning when fetching guestinfo in `early-boot-config` ([#2732])

## Build Changes

> * Treat alisa warning as errors ([#2730])

Reviewer: ```suggestion
* Treat alias warning as errors ([#2730])
```

> * Treat alisa warning as errors ([#2730])

Reviewer 4: typo: alias
```suggestion
* Treat alias warning as errors ([#2730])
```

> * Turn `source_date_epoch_from_changelog` off, removing "missing changelog" warning in build ([#2730])

Reviewer: nit: edit for length:
```suggestion
* Suppress "missing changelog" warning in build ([#2730])
```

# v 1.12.0 (2022-01-24)

## OS Changes

* Disable strict aliasing for c-utf-8 library strict aliasing in dbus-broker ([#2730])
* Add `/sys/firmware` to privileged mounts in host-ctr ([#2714])
* Use user-provided registry credentials for public.ecr.aws in host-ctr ([#2676])
* Build masked paths list dynamically in host-ctr ([#2637])
* Enable EFI option in systemd ([#2714])
* Allow simple enums as map keys in datastore ([#2687])

> * Add method for fetching hostname in imdsclient ([#2647])

Reviewer 2: I might modify this (or add another item) to call out the stated goal of [the PR this was attached to](https://github.com/bottlerocket-os/bottlerocket/pull/2647):

e.g. "Improve reliability of `settings.network.hostname` generator"
